src/wallet/index.ts

```diff
@@ -0,0 +1,92 @@
+import type { Account, Address, Chain, Transport } from "viem";
+import type { GelatoWalletClient } from "../actions";
+
+export enum WalletType {
+  Gelato = "gelato",
+  Kernel = "kernel",
+  Safe = "safe"
+}
+
+export enum WalletEncoding {
+  ERC7821 = "erc7821",
+  Safe = "safe"
+}
+
+export interface Delegation {
+  address: Address;
+  authorized: boolean;
+}
+
+export interface Gelato {
+  readonly type: WalletType.Gelato;
+  readonly encoding: WalletEncoding.ERC7821;
+  readonly isViaEntryPoint: boolean;
+  readonly eip7702: boolean;
+}
+
+export interface Kernel {
+  readonly type: WalletType.Kernel;
+  readonly encoding: WalletEncoding.ERC7821;
+  readonly isViaEntryPoint: boolean;
+  readonly eip7702: boolean;
+}
+
+export interface Safe {
+  readonly type: WalletType.Safe;
+  readonly encoding: WalletEncoding.Safe;
+  readonly isViaEntryPoint: boolean;
+  readonly eip7702: boolean;
+}
```

Comment:
```suggestion
export interface BaseWallet {
  readonly type: WalletType;
  readonly encoding: WalletEncoding;
  readonly isViaEntryPoint: boolean;
  readonly eip7702: boolean;
}

export interface Gelato extends BaseWallet {
  readonly type: WalletType.Gelato;
}

export interface Kernel extends BaseWallet {
  readonly type: WalletType.Kernel;
}

export interface Safe extends BaseWallet {
  readonly type: WalletType.Safe;
  readonly encoding: WalletEncoding.Safe;
}
```
how about extending a base interface?